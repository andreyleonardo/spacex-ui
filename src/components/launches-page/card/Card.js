import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import moment from 'moment';
import './Card.scss';

function Card({ data, onClick }) {
  return (
    <div className="Card row between-xs middle-xs" onClick={onClick}>
      <div className="col-xs">
        <p><span className="Card__mission_name">Mission: {data.mission_name}</span></p>
        <p><span className="Card__launch_date">Date: {moment(data.launch_date_local).format('DD/MM/YYYY HH:mm')}</span></p>
      </div>
      <div className={classnames('col-xs Card__launch_status', { 'Card__launch_status--failed': !data.launch_success })}>
        <p>{data.launch_success ? 'Success' : 'Failed'}</p>
      </div>
    </div>
  );
}

Card.propTypes = {
  data: PropTypes.shape({
    mission_name: PropTypes.string,
    launch_date_local: PropTypes.string
  }).isRequired
}

export default Card;
