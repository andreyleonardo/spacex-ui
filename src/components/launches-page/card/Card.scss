.Card {
  cursor: pointer;
  background-color: white;
  margin: auto;
  margin-bottom: 5px;
  text-align: left;
  padding: .625em;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 1px rgba(0,0,0,0.2);

  &__mission_name {
    font-weight: bold;
  }

  &__launch_date {
    font-size: 14px;
  }

  &__launch_status {
    text-align: right;
    color: green;
    margin-right: 16px;

    &--failed {
      color: red;
    }
  }
}
.Card:hover {
  background-color: #ededed;
}
