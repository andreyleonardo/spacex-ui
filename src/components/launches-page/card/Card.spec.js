import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import { mockDate, clearMock } from '../../../../test/utils/mock-date';

describe('Card', () => {
  beforeAll(() => {
    mockDate();
  });
  afterAll(() => {
    clearMock();
  });

  it('renders without crash', () => {
    const card = shallow(<Card data={{}}/>);
    expect(card).toMatchSnapshot();
  });
});
