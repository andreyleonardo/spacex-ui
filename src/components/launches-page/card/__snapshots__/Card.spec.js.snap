// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs"
        >
          <p>
            <span
              className="Card__mission_name"
            >
              Mission: 
            </span>
          </p>
          <p>
            <span
              className="Card__launch_date"
            >
              Date: 
              09/10/2018 21:00
            </span>
          </p>
        </div>,
        <div
          className="col-xs Card__launch_status Card__launch_status--failed"
        >
          <p>
            Failed
          </p>
        </div>,
      ],
      "className": "Card row between-xs middle-xs",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <span
                className="Card__mission_name"
              >
                Mission: 
              </span>
            </p>,
            <p>
              <span
                className="Card__launch_date"
              >
                Date: 
                09/10/2018 21:00
              </span>
            </p>,
          ],
          "className": "col-xs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="Card__mission_name"
              >
                Mission: 
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Mission: ",
                  undefined,
                ],
                "className": "Card__mission_name",
              },
              "ref": null,
              "rendered": Array [
                "Mission: ",
                undefined,
              ],
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="Card__launch_date"
              >
                Date: 
                09/10/2018 21:00
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date: ",
                  "09/10/2018 21:00",
                ],
                "className": "Card__launch_date",
              },
              "ref": null,
              "rendered": Array [
                "Date: ",
                "09/10/2018 21:00",
              ],
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Failed
          </p>,
          "className": "col-xs Card__launch_status Card__launch_status--failed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Failed",
          },
          "ref": null,
          "rendered": "Failed",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs"
          >
            <p>
              <span
                className="Card__mission_name"
              >
                Mission: 
              </span>
            </p>
            <p>
              <span
                className="Card__launch_date"
              >
                Date: 
                09/10/2018 21:00
              </span>
            </p>
          </div>,
          <div
            className="col-xs Card__launch_status Card__launch_status--failed"
          >
            <p>
              Failed
            </p>
          </div>,
        ],
        "className": "Card row between-xs middle-xs",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <span
                  className="Card__mission_name"
                >
                  Mission: 
                </span>
              </p>,
              <p>
                <span
                  className="Card__launch_date"
                >
                  Date: 
                  09/10/2018 21:00
                </span>
              </p>,
            ],
            "className": "col-xs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="Card__mission_name"
                >
                  Mission: 
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Mission: ",
                    undefined,
                  ],
                  "className": "Card__mission_name",
                },
                "ref": null,
                "rendered": Array [
                  "Mission: ",
                  undefined,
                ],
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="Card__launch_date"
                >
                  Date: 
                  09/10/2018 21:00
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    "09/10/2018 21:00",
                  ],
                  "className": "Card__launch_date",
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  "09/10/2018 21:00",
                ],
                "type": "span",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Failed
            </p>,
            "className": "col-xs Card__launch_status Card__launch_status--failed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Failed",
            },
            "ref": null,
            "rendered": "Failed",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
